src/responsive-type/style.css

```diff
@@ -0,0 +1,310 @@
+    --tab-size: 2;
+    --tab: calc(var(--tab-size, 2) * 1ch);
+    line-height: 1.5;
+    tab-size: var(--tab-size);
+
+    /* musical scale ratios */
+    --minor-second: calc(16/15);
+    --major-second: calc(9/8);
+    --minor-third: calc(6/5);
+    --major-third: calc(5/4);
+    --perfect-fourth: calc(4/3);
+    --augmented-fourth: sqrt(2);
+    --perfect-fifth: calc(3/2);
+    --major-sixth: calc(5/3);
+
+    /* the ‘golden’ ratio */
+    --golden-ratio: calc((1 + sqrt(5)) / 2);
+  }
+
+  /* for re-setting the type scale within a container */
+  [type-container] {
+    container: type / inline-size;
+  }
+
+  body,
+  [type-set] {
+    --medium: clamp(1rem, 17px + 0.24cqi, 1.25rem);
+    font-size: var(--type-set-size, var(--medium));
+
+    --xx-small: calc(1rem * pow(var(--scale), -0.75));
+    --x-small: calc(1rem * pow(var(--scale), -0.5));
+    --small: calc(1rem * pow(var(--scale), -0.25));
+    --large: calc(var(--medium) * pow(var(--scale), 1));
+    --x-large: calc(var(--medium) * pow(var(--scale), 2));
+    --xx-large: calc(var(--medium) * pow(var(--scale), 3));
+    --xxx-large: calc(var(--medium) * pow(var(--scale), 4));
+
+    --scale: var(--minor-third);
+
+    @container (inline-size > 50em) {
+      --scale: var(--perfect-fourth);
+    }
+  }
+
+  /* The body doesn't have a container */
+  body {
+    @media (width > 50em) {
+      --scale: var(--perfect-fourth);
+    }
+  }
+
+  [type-set=small] {
+    --type-set-size: var(--small);
+  }
+
+  [type-set=large] {
+    --type-set-size: var(--large);
+  }
+
+  /* prose */
+  h1 {
+    font-size: var(--xx-large);
+  }
+
+  h2 {
+    font-size: var(--x-large);
+  }
+
+  h3 {
+    font-size: var(--large);
+  }
+
+  h4 {
+    font-size: var(--medium);
+  }
+
+  h1,
+  h2,
+  h3 {
+    line-height: 1.25;
+    margin-block: 0.25rlh;
+  }
+
+  p,
+  li {
+    text-wrap: pretty;
+  }
+
+  blockquote {
+    border-inline-start: medium solid;
+    font-size: var(--large);
+    font-style: italic;
+    padding-inline-start: var(--tab-size);
+  }
+
+  figcaption {
+    font-size: var(--small);
+    font-style: italic;
+  }
+}
+
+@layer layout {
+  body {
+    display: grid;
+    gap: var(--gap);
+    padding-block: var(--gap);
+
+    > :is(header, main, aside, footer) {
+      container: layout / inline-size;
+    }
+  }
+
+  main {
+    display: grid;
+    gap: 1lh 0;
+    grid-template-columns:
+      [full-start] var(--margin, var(--gap)) [pull-start] minmax(0, 1.5fr) [content-start] minmax(auto, 35em) [content-end] minmax(0, 1fr) [pull-end] var(--margin, var(--gap)) [full-end];
+
+    @media (width > 45em) {
+      --margin: 0;
+      column-gap: var(--gap);
+    }
+
+    >* {
+      margin: unset;
+      grid-column: content;
+    }
+  }
+
+  .by-line {
+    font-size: var(--small);
+    line-height: 1rlh;
+
+    @container (width > 50em) {
+      grid-column: pull-start / content-start;
+      text-align: right;
+    }
+
+    p {
+      margin: unset;
+    }
+  }
+
+  main figure {
```

Comment:
does this require main here?

Reply:
changed this to just `figure` in https://github.com/oddbird/baseline-demos/pull/10/changes/e33f41897cfcd4c78e62491defc9174b95e2cf74